File: MobileApplications/server/server/src/main/java/ro/ubbcluj/maserver/model/Dog.java
package ro.ubbcluj.maserver.model;


import lombok.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@ToString(callSuper = true)
@Builder
@Entity
public class Dog {
    @Id
    @GeneratedValue
    private Long ID;
    private String name;
    private String race;
    private String description;
    private int age;

}

File: MobileApplications/native/DogShelterManagement/app/src/main/java/ro/ubbcluj/dogsheltermanagement/model/Dog.kt
package ro.ubbcluj.dogsheltermanagement.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class Dog(
    @PrimaryKey(autoGenerate = false) val id: Long?,
    @ColumnInfo(name = "name") val name: String,
    @ColumnInfo(name = "race") val race: String,
    @ColumnInfo(name = "description") val description: String,
    @ColumnInfo(name = "age") val age: Int
){
    constructor(name: String, race: String, description: String, age: Int) : this(null,name,race,description,age)

    override fun equals(other: Any?): Boolean {
        if(other is Dog){
            return other.id == this.id
        }
        return false;
    }
}